### ISSUE TITLE
Breaking change (ish): lint_package() fails if loadNamespace() doesn't work for some deps

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1360]
Finished a run of #1304 thanks to #1348 and am peeking at results now.

The first thing I'm noticing is that `lint_package()` fails on a lot of repos where it still succeeds on v2.0.1, in particular with error messages like:

```
[babette]: there is no package called ‘beastier’
[BiasCorrector] there is no package called ‘shiny’
[BTYDplus]: there is no package called ‘BTYD’
[caretEnsemble]: there is no package called ‘caret’
[cloudos]: there is no package called ‘ggplot2’
```

Haven't investigated why, but if this breaking change is intentional, we should mention it (I don't recall anything in the NEWS)

**Comment by MichaelChirico** [comment]
Looks like the updates to `object_name_linter()` and `object_length_linter()` are to blame -- should we offer an option for these that would skip `loadNamespace()` attempts to smooth the transition?

In any case I'll have to update the script to attempt to install required packages first...

**Comment by AshesITR** [comment]
I actually consider this a bug. We should always lint what we can, not choke on incomplete dependencies.

**Comment by MichaelChirico** [comment]
that was my first thought as well... should we throw a warning though? to advise the user they might avoid false positives (no false negatives, right?) by installing what's missing? maybe just put something in the lint message?

**Comment by AshesITR** [comment]
Hmm maybe a warning, but I'd leave the lint message as-is.
I think only false positives regarding unknown function names or generics are possible, so it should be fine.

**Comment by AshesITR** [comment]
It just occurred to me that when linting many files, the linters will potentially generate a ton of the same type of warning. So maybe it's better to stay silent...

**Comment by MichaelChirico** [comment]
Would be great if we could emit the warning at "compile time" when the factory is run, not sure if that's feasible

**Comment by AshesITR** [comment]
That would work only if we changed our design decision to prepare export lists etc. at compile time.
